processing/src/main/java/io/druid/segment/filter/SpatialFilter.java

```diff
@@ -58,9 +61,15 @@ public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
     return factory.makeValueMatcher(
         dimension,
-        new Predicate()
+        new DruidCompositePredicate()
         {
           @Override
+          public boolean applyLong(long value)
+          {
+            throw new UnsupportedOperationException("Spatial filter is not supported on long columns.");
```

Comment:
better to `return false`.

OR treat the high 32 bits as the x coordinate and the low 32 bits as the y coordinate. Easter egg!